### PR TITLE
Remove transport abstraction

### PR DESCRIPTION
**Pull request by Noarkhh**
This PR removes the RTSP.Transport behavior, which was essentially only implemented by an abstraction over a TCP socket.

### CODE REVIEW

File: lib/membrane_rtsp/session/logic.ex
Comment (mat-hek) [inline lib/membrane_rtsp/session/logic.ex:1]: let's merge this file with `rtsp.ex`

File: lib/membrane_rtsp/session/logic.ex [hunk 1]
```diff
@@ -2,12 +2,12 @@ defmodule Membrane.RTSP.Logic do
   @moduledoc """
   Logic for RTSP session
   """
-  alias Membrane.RTSP.{Request, Response}
+  alias Membrane.RTSP.{Request, Response, TCPSocket}
   @user_agent "MembraneRTSP/#{Mix.Project.config()[:version]} (Membrane Framework RTSP Client)"
 
   defmodule State do
     @moduledoc "Struct representing the state of RTSP session"
```
Comment (mat-hek) [inline lib/membrane_rtsp/session/logic.ex]: this should have `@moduledoc false`

File: lib/membrane_rtsp/tcp_socket.ex [hunk 2]
```diff
@@ -1,31 +1,23 @@
-defmodule Membrane.RTSP.Transport.TCPSocket do
+defmodule Membrane.RTSP.TCPSocket do
   @moduledoc """
```
Comment (mat-hek) [inline lib/membrane_rtsp/tcp_socket.ex]: it should be `@moduledoc false`

File: lib/membrane_rtsp/rtsp.ex [hunk 3]
```diff
@@ -22,21 +22,20 @@ defmodule Membrane.RTSP do
     * options - a keyword list that shall be passed when executing request over
     transport
   """
-  @spec start_link(binary(), module() | URI.t(), Keyword.t()) :: GenServer.on_start()
-  def start_link(url, transport \\ Membrane.RTSP.Transport.TCPSocket, options \\ []) do
-    do_start(url, transport, options, &GenServer.start_link/2)
+  @spec start_link(binary() | URI.t(), Keyword.t()) :: GenServer.on_start()
```
Comment (mat-hek) [inline lib/membrane_rtsp/rtsp.ex]: let's add a better typespec for options

File: lib/membrane_rtsp/rtsp.ex [hunk 4]
```diff
@@ -22,21 +22,20 @@ defmodule Membrane.RTSP do
     * options - a keyword list that shall be passed when executing request over
```
Comment (mat-hek) [inline lib/membrane_rtsp/rtsp.ex]: remote transport from the doc

File: test/membrane_rtsp/session/session_logic_test.exs [hunk 5]
```diff
@@ -31,22 +32,26 @@ defmodule Membrane.RTSP.SessionLogicTest do
...
       state: state
     } do
-      resolver = fn _request -> {:error, :timeout} end
-      state = %State{state | execution_options: [resolver: resolver]}
+      mock(:gen_tcp, [send: 2], fn _socket, _request ->
+        {:error, :timeout}
+      end)
+
+      # resolver = fn _request -> {:error, :timeout} end
+      # state = %State{state | execution_options: [resolver: resolver]}
```
Comment (mat-hek) [inline test/membrane_rtsp/session/session_logic_test.exs]: leftover

File: test/membrane_rtsp/session/session_logic_test.exs [hunk 6]
```diff
@@ -90,74 +95,77 @@ defmodule Membrane.RTSP.SessionLogicTest do
...
   test "add digest information in the state", %{state: state, request: request} do
-    resolver = fn _request ->
+    mock(:gen_tcp, [send: 2], fn _socket, _request ->
       {:ok,
        "RTSP/1.0 200 OK\r\nWWW-Authenticate: Digest realm=\"realm\", nonce=\"nonce\"\r\n\r\n"}
-    end
+    end)
 
-    state = %State{state | execution_options: [resolver: resolver]}
+    # state = %State{state | execution_options: [resolver: resolver]}
```
Comment (mat-hek) [inline test/membrane_rtsp/session/session_logic_test.exs]: leftover

File: lib/membrane_rtsp/tcp_socket.ex [hunk 7]
```diff
@@ -1,31 +1,23 @@
...
+  This module transmits requests over TCP
   Socket keeping connection until either session is closed or connection is
   closed by server.
 
   Supported options:
     * timeout - time after request will be deemed missing and error shall be
      returned.
   """
-  use Membrane.RTSP.Transport
   import Mockery.Macro
```
Comment (mat-hek) [inline lib/membrane_rtsp/tcp_socket.ex:3]: Probably can be removed here and from deps

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane_rtsp/rtsp.ex [hunk 8]
```diff
@@ -1,50 +1,141 @@
...
 
+  @type options() :: [option()]
+  @type option() ::
+          {:connection_timeout, non_neg_integer()} | {:response_timeout, non_neg_integer()}
+
   @default_rtsp_port 554
+  @user_agent "MembraneRTSP/#{Mix.Project.config()[:version]} (Membrane Framework RTSP Client)"
+
+  defmodule State do
+    @moduledoc "Struct representing the state of RTSP session"
```
Comment (mat-hek) [inline lib/membrane_rtsp/rtsp.ex]: Should be `@moduledoc false`. Generate the docs and make sure that all and only public API is there. I think we can also remove `Membrane.RTSP.Server.Logic` and `Membrane.RTSP.Server.Logic.State` from there

File: lib/membrane_rtsp/tcp_socket.ex (hunk 7 above)
Comment (mat-hek) [reply lib/membrane_rtsp/tcp_socket.ex:3]: bump @Noarkhh

Comment (Noarkhh) [reply lib/membrane_rtsp/tcp_socket.ex:3]: Do you mean Mockery? It's still used in tests

Comment (mat-hek) [reply lib/membrane_rtsp/tcp_socket.ex:3]: Ok, somehow I thought it's removed

**Review by mat-hek — APPROVED** [review]

**Pull request merged**